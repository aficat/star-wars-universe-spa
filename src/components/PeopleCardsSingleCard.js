import React, { Component } from 'react';
import { Card, CardActionArea, CardContent, Typography, Grid, Tooltip } from '@material-ui/core';
import { Modal } from 'semantic-ui-react';

const cardStyles = {
    textAlign: "left"
};

class PeopleCardsSingleCard extends Component {

    constructor(props) {
        super(props);
        this.state = {
            open: false,
        }
    }

    renderPersonCard = () => {
        const { name, height, mass, gender, eyeColor, skinColor, hairColor, birthYear } = this.props;
        return (
            <Card style={cardStyles}>
                <CardActionArea onClick={() => this.handleClick()}>
                    <CardContent>
                        <Typography
                            variant="h5"
                            component="h2"
                        >
                            {name}
                        </Typography>
                        <Typography
                            variant="subtitle2"
                            component="p"
                        >
                            Gender: {gender}<br />
                        Birth Year: {birthYear}
                        </Typography>
                        <Typography
                            variant="body2"
                            component="p"
                        >
                            Height: {height}<br />
                            Mass: {mass}<br />
                            Eye Color: {eyeColor} <br />
                            Skin Color: {skinColor} <br />
                            Hair Color: {hairColor}
                            <br />
                        </Typography>
                    </CardContent>
                </CardActionArea>
            </Card>
        );
    }

    handleClick = () => {
        this.setState({ open: true })
    };

    handleClose = () => {
        this.setState({ open: false })
    };

    renderPersonCardModal = () => {
        const { name, height, mass, gender, url, eyeColor, skinColor, hairColor, birthYear } = this.props;
        return (
            <Modal size={"small"} open={this.state.open} onClose={this.handleClose} closeIcon>
                <Modal.Header>{name}</Modal.Header>
                <Modal.Content scrolling>
                    <Typography
                        variant="subtitle2"
                        component="p"
                    >
                        Gender: {gender}<br />
                        Birth Year: {birthYear}
                    </Typography>
                    <Typography
                        variant="body2"
                        component="p"
                    >
                        Height: {height}<br />
                            Mass: {mass}<br />
                            Eye Color: {eyeColor} <br />
                            Skin Color: {skinColor} <br />
                            Hair Color: {hairColor}
                        <br />
                    </Typography>
                </Modal.Content>
            </Modal>
        )
    }

    render() {
        return (
            <Grid item xs={12} md={3}>
                <Tooltip
                    title="Click to view more details."
                    placement="bottom"
                    arrow
                >
                    {this.renderPersonCard()}
                </Tooltip>
                {this.renderPersonCardModal()}
            </Grid>
        );
    }
}

export default PeopleCardsSingleCard;
