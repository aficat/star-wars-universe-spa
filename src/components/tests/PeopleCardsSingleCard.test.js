import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import toJson from 'enzyme-to-json'
import PeopleCardsSingleCard from '../PeopleCardsSingleCard';
import store from '../../store';

describe('PeopleCardsSingleCard component', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<PeopleCardsSingleCard store={store} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders without crashing given the required props', () => {
        const props = {
            dispatch: jest.fn(),
            homeworld: "",
            species: [],
            films: [],
            starships: [],
            vehicles: [],
        }
        const wrapper = shallow(<PeopleCardsSingleCard {...props} store={store} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

});