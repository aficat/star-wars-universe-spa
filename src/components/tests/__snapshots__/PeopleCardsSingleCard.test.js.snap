// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleCardsSingleCard component renders without crashing given the required props 1`] = `
<ContextProvider
  value={null}
>
  <PeopleCardsSingleCard
    dispatch={[MockFunction]}
    fetchFilms={[Function]}
    fetchHomeworld={[Function]}
    fetchSpecies={[Function]}
    fetchStarships={[Function]}
    fetchVehicles={[Function]}
    films={Array []}
    homeworld=""
    species={Array []}
    starships={Array []}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    vehicles={Array []}
  />
</ContextProvider>
`;
